python/tempo/tsdf.py

```diff
@@ -150,10 +150,12 @@ def __getLastRightRow(self, left_ts_col, right_cols, sequence_col, tsPartitionVa
     if tsPartitionVal is not None:
       for column in df.columns:
         if (column.startswith("non_null")):
-          any_blank_vals = (df.agg({column: 'min'}).collect()[0][0] == 0)
-          newCol = column.replace("non_null_ct", "")
-          if any_blank_vals:
-            logger.warning("Column " + newCol + " had no values within the lookback window. Consider using a larger window to avoid missing values. If this is the first record in the data frame, this warning can be ignored.")
+          # Avoid collect() calls when explicitly ignoring the warnings about null values due to lookback window.
+          if not suppress_null_warning and logger.isEnabledFor(logging.WARNING):
+            any_blank_vals = (df.agg({column: 'min'}).collect()[0][0] == 0)
```

Comment:
The original request actually does the collect if the WARNING level is enabled. Switch 'and' to 'or'.

Reply:
Changed the condition from 'and' to 'or'